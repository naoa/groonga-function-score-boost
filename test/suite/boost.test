register functions/score_boost

table_create Entries TABLE_NO_KEY
column_create Entries title COLUMN_SCALAR ShortText

load --table Entries
[
{"title": "groonga and MySQL"}
]

select Entries --filter 'all_records() && score_boost(10.0)' --output_columns '_id,_score,title'
