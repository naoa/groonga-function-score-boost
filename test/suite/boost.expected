register functions/score_boost
[[0,0.0,0.0],true]
table_create Entries TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Entries title COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
load --table Entries
[
{"title": "groonga and MySQL"}
]
[[0,0.0,0.0],1]
select Entries --filter 'all_records() && score_boost(10.0)' --output_columns '_id,_score,title'
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "_score",
          "Int32"
        ],
        [
          "title",
          "ShortText"
        ]
      ],
      [
        1,
        10,
        "groonga and MySQL"
      ]
    ]
  ]
]
